### DISCUSSION TITLE
How to implement single entry point (main component) in Nuxt2 (like app.vue in Nuxt3)

### DISCUSSION THREAD

**Discussion by bagaskarala** [nuxt/nuxt#19935] [Questions (Nuxt 2)]
I want to initiate global element and function which is only run once on first load and in all pages. But i can't find any single entry point in nuxt2. I have many layouts so it is not reliable to declare the function in each layout. Is there any workaround to handle this case?

TLDR, how to `app.vue` like in [nuxt3](https://nuxt.com/docs/guide/directory-structure/app) to [work in every route](https://nuxt.com/docs/getting-started/views#appvue)

**Comment by warflash** [comment]
Not sure if that's a typo in your question but nuxt 3 also has an app.vue you might be able to use: https://nuxt.com/docs/guide/directory-structure/app

**Reply by bagaskarala** [reply]
@warflash what things i need is to implement single global `app.vue` in nuxt2
